### PR TITLE
builtin: add elt built-in function

### PR DESCRIPTION
**Pull request by hsqlu**
[Add the ELT built-in string function.](https://github.com/pingcap/tidb/issues/2863)

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
@hsqlu 
Thanks for your PR.
If possible, please check and register the builtin function you determine to implement in [issue](https://github.com/pingcap/tidb/issues/2863) to let others know, thank you!

**Comment by hsqlu** [comment]
@XuHuaiyu OK. I will create a issue to mention that.

**Comment by coocood** [comment]
LGTM

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1557,7 +1557,28 @@ type builtinEltSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.6/en/string-functions.html#function_elt
 func (b *builtinEltSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("elt")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
```
Comment (zimulala) [inline expression/builtin_string.go]: Return `d, errors.Trace(err)` is better.

Comment (hsqlu) [inline expression/builtin_string.go]: Yes. That would be unified. I've modified. 
PTAL. Thanks!

**Comment by zimulala** [comment]
LGTM

**Pull request merged**